### ISSUE TITLE
Update `DashboardGoalsWidget` to be inactive in view-only context if there are no goals

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#4817]
## Feature Description

The dashboard goals widget is only relevant on the shared dashboard when there is data to show.

When there are no goals, this widget shows a CTA to create your first goal which isn't something a dashboard viewer would be able to do in this context. Accordingly, the widget should be hidden in this case.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The "Use goals to measure success" CTA rendered by the `DashboardGoalsWidget` should not be rendered in the context of the view-only/shared dashboard.
	* In this situation, the widget should not appear at all (i.e. rendering `WidgetNull`)

## Implementation Brief
* Using `assets/js/modules/analytics/components/dashboard/DashboardGoalsWidget.js`,
	* Add `WidgetNull` to the list of props for the component.
	* Use the `useViewOnly` hook to determine whether we are in view only mode.
	* If it returns `true`, render `WidgetNull` instead of `CTA` when there are no goals.
* Create `assets/js/modules/analytics/components/dashboard/DashboardGoalsWidget.stories.js` which will contain stories ported from `stories/module-analytics-components.stories.js` for `DashboardGoalsWidget`, i.e 
	* `Loaded`
	* `Loading`
	* `Data Unavailable`
	* `Zero Data`
	* `No Goals`
	* `Error`
	* `Gathering Data` new story
	* `View Only` new story

### Test Coverage
* No new tests to be added.

## QA Brief

* Ensure the existing Analytics Goals widget functions as-expected.
* Check https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-analytics-widgets-dashboardgoalswidget--no-goals-set-view-only and ensure it outputs nothing.

## Changelog entry

* Hide Analytics goals CTA in view-only context.

**Comment by eugene-manuilov** [comment]
IB ✔️

**Comment by tofumatt** [comment]
@asvinb I'm a bit confused by the IB section for the stories: did you mean to simply move the file over or to actually port all of the stories over to V6-type stories?

I'm finding it a rear bear to work with these older `generateReportBasedWidgetStories`—is there an example of a conversion you can point me to? I find them really hard to parse and I think my brain is just getting a bit stuck on how to handle the storybook part of the issue 😓

**Comment by asvinb** [comment]
@tofumatt I meant to port all the stories for `DashboardGoalsWidget` over to the v6 type stories. I thought it's a good way to gradually phase out the old way of creating stories. I've set up a branch where I've ported the stories over:
https://github.com/google/site-kit-wp/blob/ib/4817-stories/assets/js/modules/analytics/components/dashboard/DashboardGoalsWidget.stories.js
I used `assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js` as starting point. Basically we just need to replace the report options from what we have in the old stories.

**Comment by tofumatt** [comment]
No, the existing widgets should be tested _without_ Dashboard Sharing enabled to ensure they still work as-expected when dashboard sharing is **not enabled**.

Huh, I'm not sure why that is redirecting but I've updated the link in the QA Brief… if it still doesn't work try entering the URL manually? Or searching for the "Analytics > Widgets > Dashboard Goals Widgets > No Goals Set (View-only)" story here:

<img width="243" alt="CleanShot 2022-04-27 at 15 35 09" src="https://user-images.githubusercontent.com/90871/165543248-9ee9eef9-7a83-4c27-a6ea-29a3c5845da2.png">

**Comment by aaemnnosttv** [comment]
The URL was redirecting because it is missing the leading slash before `story` in the `path` query parameter. This works: https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-analytics-widgets-dashboardgoalswidget--no-goals-set-view-only

**Comment by mohitwp** [comment]
### QA Update ✅ 

**Verified**

- Analytics goals widget working as-expected when dashboard sharing is not enabled.

- Also, verified when zero data states feature is active for both gathering and zero data state.

- Verified Storybook - https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-analytics-widgets-dashboardgoalswidget--no-goals-set-view-only - It outputs nothing.


<details>

![image](https://user-images.githubusercontent.com/94359491/165742470-e88cd2a8-b264-4ee7-bbf5-7892fca3ce5f.png)




![image](https://user-images.githubusercontent.com/94359491/165742078-ac595170-5fd0-4ade-99f0-c84675b69f85.png)

![image](https://user-images.githubusercontent.com/94359491/165742112-410268ce-720d-49d6-8581-11f5481bf85a.png)

![image](https://user-images.githubusercontent.com/94359491/165742138-743719fd-2e07-4e24-af72-7cbe3e7b8431.png)

</details>


-
